src/basho_bench_worker_sup.erl

```diff
@@ -71,6 +91,23 @@ worker_specs(0, Acc) ->
     Acc;
 worker_specs(Count, Acc) ->
     Id = list_to_atom(lists:concat(['basho_bench_worker_', Count])),
-    Spec = {Id, {basho_bench_worker, start_link, [Id, Count]},
-            transient, 5000, worker, [basho_bench_worker]},
+    %% Use "single_worker" atom for original non-worker case
+    Spec = {Id, {basho_bench_worker, start_link, [Id, {single_worker, Count, Count}, []]},
+                 transient, 5000, worker, [basho_bench_worker]},
     worker_specs(Count-1, [Spec | Acc]).
+
+worker_specs_multi([], _BaseGlobalId, Acc) ->
+    Acc;
+worker_specs_multi([{WorkerType, Count, Conf} | Rest], BaseGlobalId, Acc0) ->
+    Acc = lists:foldl(
+        fun(I, AccP) ->
+            Id = list_to_atom(lists:concat(
+                ['basho_bench_worker_', WorkerType, '_', I])),
+            Spec = {
+                Id,
+                {basho_bench_worker, start_link, [Id, {WorkerType, I, I+BaseGlobalId}, Conf]},
+                transient, 5000, worker, [basho_bench_worker]},
+            AccP ++ [Spec]
```

Comment:
Concatenating lists like this is usually not what you want, as it's quite inefficient. You can flip this around and do `[Spec | AccP]` which is quite efficient. The only time you should use `++` is for combining strings (and even then don't do it in a tight loop).